File: Algorithms/Implementation/Taum and B'day.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;

class Result
{

    /*
     * Complete the 'taumBday' function below.
     *
     * The function is expected to return a LONG_INTEGER.
     * The function accepts following parameters:
     *  1. INTEGER b
     *  2. INTEGER w
     *  3. INTEGER bc
     *  4. INTEGER wc
     *  5. INTEGER z
     */

    public static long taumBday(int b, int w, int bc, int wc, int z)
    {
        long result = 0;
        long costB = Math.Min(bc, wc + z);
        long costW = Math.Min(wc, bc + z);

        result = b * costB + w * costW;

        return result;
    }
}

class Solution
{
    public static void Main(string[] args)
    {
        TextWriter textWriter = new StreamWriter(@System.Environment.GetEnvironmentVariable("OUTPUT_PATH"), true);

        int t = Convert.ToInt32(Console.ReadLine().Trim());

        for (int tItr = 0; tItr < t; tItr++)
        {
            string[] firstMultipleInput = Console.ReadLine().TrimEnd().Split(' ');
            int b = Convert.ToInt32(firstMultipleInput[0]);
            int w = Convert.ToInt32(firstMultipleInput[1]);

            string[] secondMultipleInput = Console.ReadLine().TrimEnd().Split(' ');
            int bc = Convert.ToInt32(secondMultipleInput[0]);
            int wc = Convert.ToInt32(secondMultipleInput[1]);
            int z = Convert.ToInt32(secondMultipleInput[2]);

            long result = Result.taumBday(b, w, bc, wc, z);

            textWriter.WriteLine(result);
        }

        textWriter.Flush();
        textWriter.Close();
    }
}

File: Algorithms/Implementation/Taum and B'day.php
<?php

/*
 * Complete the 'taumBday' function below.
 *
 * The function is expected to return a LONG_INTEGER.
 * The function accepts following parameters:
 *  1. INTEGER b
 *  2. INTEGER w
 *  3. INTEGER bc
 *  4. INTEGER wc
 *  5. INTEGER z
 */

function taumBday($b, $w, $bc, $wc, $z) {
    # Write your code here
    $result = 0;
    $result = $b * $bc + $w * $wc;

    $result = min($result, $b * ($wc + $z) + $w * $wc, $b * $bc + $w * ($bc + $z));

    return $result;
}

$fptr = fopen(getenv("OUTPUT_PATH"), "w");

$t = intval(trim(fgets(STDIN)));

for ($t_itr = 0; $t_itr < $t; $t_itr++) {
    $first_multiple_input = explode(' ', rtrim(fgets(STDIN)));
    $b = intval($first_multiple_input[0]);
    $w = intval($first_multiple_input[1]);

    $second_multiple_input = explode(' ', rtrim(fgets(STDIN)));
    $bc = intval($second_multiple_input[0]);
    $wc = intval($second_multiple_input[1]);
    $z = intval($second_multiple_input[2]);

    $result = taumBday($b, $w, $bc, $wc, $z);

    fwrite($fptr, $result . "\n");
}

fclose($fptr);
